### PR TITLE
The login functionality is performed with its respective validations

### PR DESCRIPTION
**Pull request by Elvis-Mieles**
## summary
The login functionality is performed with its respective validations

### CODE REVIEW

File: src/test/java/training/sign_in/SingIn.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package training.sign_in;
+
+import com.intuit.karate.junit5.Karate;
+
+public class SingIn {
```
Comment (jmzagowolox) [inline src/test/java/training/sign_in/SingIn.java]: the correct "word" would be "SignIn"

File: src/test/java/training/sign_in/responsesuccessfully.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "user": {
+    "id": "#number",
+    "email": "#string",
+    "username": "#string",
+    "bio": "#null",
```
Comment (jmzagowolox) [inline src/test/java/training/sign_in/responsesuccessfully.json]: is this always null? if not, add the data type with double "#" in order to indicate this field is optional

File: src/test/java/training/sign_in/responsesuccessfully.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+    "image": "#null",
```
Comment (jmzagowolox) [inline src/test/java/training/sign_in/responsesuccessfully.json]: the same for bio

File: src/test/java/training/sign_in/sing_in.feature [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    * def emailNotExist = emailNotExist
+    * def passwordIncorrect = passwordIncorrect
+
+  @flow
+  Scenario: Login successfully
+
+    * def login = {"user": {"email": '#(email)',"password": '#(password)'}}
+    * def loginSuccessfully = read('training/sign_in/responsesuccessfully.json')
+
+    Given path 'users/login'
```
Comment (jmzagowolox) [inline src/test/java/training/sign_in/sing_in.feature]: separate the routes like this 
`Given path 'users', 'login'`

File: src/test/java/training/sign_in/sing_in.feature [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    Then status 200
+    And match  response == loginSuccessfully
+    And assert response.user.username == username
+    And assert response.user.email == email
+
+  @unitTest
+  Scenario Outline: login incorrect
+
+    * def loginIncorrect = {"user": {"email": '#(<descriptionEmail>)',"password": '#(<descriptionPassword>)'}}
+    * def incorrectLogin = {"errors": {"email or password": <descriptionMessage>}}
```
Comment (jmzagowolox) [inline src/test/java/training/sign_in/sing_in.feature]: delete this variable if you wont use it.

**Review by jmzagowolox — CHANGES_REQUESTED** [review]

**Comment by Elvis-Mieles** [comment]
## summary
Pull request corrections are made

File: src/test/java/training/sign_in/sign_in.feature [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    * def emailNotExist = emailNotExist
+    * def passwordIncorrect = passwordIncorrect
+
+  @flow
+  Scenario: Login successfully
+
+    * def login = {"user": {"email": '#(email)',"password": '#(password)'}}
+    * def loginSuccessfully = read('training/sign_in/responsesuccessfully.json')
+
+    Given path 'users/login'
```
Comment (jmzagowolox) [inline src/test/java/training/sign_in/sign_in.feature]: look at line 31 and use that way

**Review by jmzagowolox — CHANGES_REQUESTED** [review]

**Comment by Elvis-Mieles** [comment]
## summary
The path correction is performed on successful login

**Comment by Elvis-Mieles** [comment]
## summary
The description of the feature is organized

**Comment by Elvis-Mieles** [comment]
## summary
The description of the feature is organized

**Review by jmzagowolox — APPROVED** [review]

**Pull request merged**